File: pset2/readability/readability.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

int runColemanLiau(string text);

int main(void)
{
    string inputText = get_string("Text: ");

    int grade = runColemanLiau(inputText);

    if (grade < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (grade >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", grade);
    }
}

int runColemanLiau(string text)
{
    int nLetters = 0;
    int nWords = 1;
    int nSentences = 0;
    // Only run once through the text
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        // The char is a letter
        if (isalpha(text[i]))
        {
            nLetters++;
        }
        // The char is any kind of space
        else if (isspace(text[i]))
        {
            nWords++;
        }
        // The char denotes end of sentence
        else if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            nSentences++;
        }
    }

    // Calculate params for the formula
    double L = (double) nLetters / nWords * 100;
    double S = (double) nSentences / nWords * 100;
    // Coleman-Liau formula
    double index = 0.0588 * L - 0.296 * S - 15.8;
    return round(index);
}

File: pset6/readability/readability.py
from cs50 import get_string


def main():
    inputText = get_string("Text: ")
    grade = runColemanLiau(inputText)

    if grade < 1:
        print("Before Grade 1")
    elif grade >= 16:
        print("Grade 16+")
    else:
        print("Grade " + str(round(grade)))


def runColemanLiau(text):
    nLetters = 0
    nWords = 1
    nSentences = 0
    # Iterate over each character in the text
    for char in text:
        # The char is a letter
        if char.isalpha():
            nLetters += 1
        # The char is any kind of space
        elif char.isspace():
            nWords += 1
        # The char denotes end of sentence
        elif char == "." or char == "!" or char == "?":
            nSentences += 1

    # Calculate params for the formula
    L = nLetters / nWords * 100
    S = nSentences / nWords * 100
    # Coleman-Liau formula
    index = 0.0588 * L - 0.296 * S - 15.8
    return index


main()